.github/workflows/ci.yml

```diff
@@ -151,8 +161,14 @@ jobs:
   # host tests
 
   host-tests:
-    runs-on: macos-m1-self-hosted
-
+    uses: ./.github/workflows/check_runner.yml
+    with:
+      primary-runner: macos-m1-self-hosted
+      fallback-runner: ubuntu-latest
+    
+  run-host-tests:
```

Comment:
Could you rename these, so that the names for the actual jobs don't change? Instead of `host-tests -> run-host-tests` call them `detect-host-tests-runner -> host-tests`, so that we don't need to update the branch protection rules.